File: solutions/go/grains/1/grains.go
package grains

import "errors"

func Square(number int) (uint64, error) {
	if number >= 1 && number <= 64 {
		return uint64(1) << (number - 1), nil
	}
	return 0, errors.New("")
}

func Total() uint64 {
	return ^uint64(0)
}

File: solutions/erlang/grains/1/src/grains.erl
-module(grains).
-export([square/1, total/0]).
-spec square(integer()) -> pos_integer() | {error, string()}.
square(X) when is_integer(X), X > 0, X =< 64 ->
    1 bsl (X - 1); % 
square(_) ->
  {error, "square must be between 1 and 64"}.
-spec(total() -> pos_integer()).
total() -> 1 bsl 64 - 1.